addon/-private/route-ext.js

```diff
@@ -62,7 +62,7 @@ Route.reopen({
 
     if (owner.routable) {
       let prefix = owner.mountPoint;
-      routeName = `${prefix}.${_routeName}`;
+      routeName =  _routeName !== 'application' ? `${prefix}.${_routeName}` : prefix;
```

Comment:
Looks good, but I'd prefer to avoid the ternary if you don't mind.

```js
if (owner.routable) {
  let prefix = owner.mountPoint;
  if (routeName === 'application') {
    routeName = prefix;
  } else {
    routeName = `${prefix}.${_routeName}`;
  }
}
```